### PR TITLE
📘 DOC: Adding github actions example to deploy.md

### PR DESCRIPTION
**Pull request by danny-burrows**
## Changes
Added an example of a Github actions workflow in the deploy.md docs for github pages that pushes changes to \<YOUR USERNAME\>.github.io.

## Testing
The yaml example code was tested on my personal repo, changes proposed here are only in the deploy.md Docs.

## Docs
Yes, I added an example under the Github Actions heading in deploy.md.

Thank you, hope this is helpful.

### CODE REVIEW

File: docs/src/pages/guides/deploy.md
Comment (jasikpark) [inline docs/src/pages/guides/deploy.md:72]:
```suggestion
2. Create the file `.github/workflows/main.yml` and add in the yaml below. Make sure to edit in your own details.
```

**Comment by jasikpark** [comment]
This looks pretty great!

File: docs/src/pages/guides/deploy.md
Comment (FredKSchott) [inline docs/src/pages/guides/deploy.md:74]: This line confused me. what does "your github source code repo" mean? maybe "the astro project repo" instead?

File: docs/src/pages/guides/deploy.md
Comment (FredKSchott) [inline docs/src/pages/guides/deploy.md:120]: one thing that could help: can you move these into variables somewhere at the top of the file, so that users have 1 place to edit values?

Also, can some of these be auto-detected, like `github.actor`? Just found this list here: https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context

**Comment by danny-burrows** [comment]
Thanks for the suggestions, I think you are definitely right with the wording of the project repo, just wanted to make sure it couldn't be confused for the deployment repo.

- I also added a env section and minimized the amount of changes needed to the script. 
- I kept the deployment repo name as input so you can still deploy to any project github page.

Thank you :)

**Review by FredKSchott — APPROVED** [review]
LGTM once you've merged the two suggestions made by @jasikpark and myself!

**Comment by danny-burrows** [comment]
I've applied the changes you & @jasikpark suggested, thank you both very much for your help.

**Comment by FredKSchott** [comment]
no problem, thanks for adding this! It's a great addition

**Pull request merged**
